### PR TITLE
 Add support for Terraform package type #354

### PR DESCRIPTION
**Pull request by nevenr**
Please review.

Thanks.

Regards,
N

### CODE REVIEW

**Comment by nevenr** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by nevenr** [comment]
recheck

File: services/src/test/groovy/org/jfrog/artifactory/client/TerraformPackageTypeRepositoryTests.groovy [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+        }
+
+        if (repositoryType == RepositoryTypeImpl.FEDERATED) {
+            def settings = new TerraformRepositorySettingsImpl()
+            settings.with {
+                terraformType = TerraformRepositorySettings.TerraformType.module
+                repoLayout = moduleLayout
+            }
+            return settings
+        }
```
Comment (yahavi) [inline services/src/test/groovy/org/jfrog/artifactory/client/TerraformPackageTypeRepositoryTests.groovy:51]: Let's use switch-case here

**Review by yahavi — APPROVED** [review]
Thanks @nevenr!

**Pull request merged**